stl/inc/atomic

```diff
@@ -728,7 +728,7 @@ struct _Atomic_storage<_Ty, 8> { // lock-free using 8-byte intrinsics
         return reinterpret_cast<_Ty&>(_As_bytes);
     }
 
-#ifdef _M_IX86
+#if defined(_M_IX86) && defined(__clang__)

```

Comment:
Both `#endif` comments also need to be updated. #351 remains active, but I'd prefer to see an arrow comment:

```cpp
#endif // ^^^ !defined(_M_IX86) || !defined(__clang__) ^^^
```